### PR TITLE
neutrino: use functional option for config passed to NewChainService

### PR DESCRIPTION
**Pull request by parakeety**
- add functional options for `Config` passed to `NewChainService`
- change sync_test.go accordingly

### CODE REVIEW

File: sync_test.go [hunk 1]
```diff
@@ -356,21 +356,11 @@ func TestSetup(t *testing.T) {
...
-			h2.P2PAddress(),
-			h1.P2PAddress(),
-		},
-	}
 
 	neutrino.MaxPeers = 3
 	neutrino.BanDuration = 5 * time.Second
 	neutrino.WaitForMoreCFHeaders = time.Second
-	svc, err := neutrino.NewChainService(config)
+	svc, err := neutrino.NewChainService(neutrino.DataDir(tempDir), neutrino.Database(db), neutrino.ChainParams(modParams), neutrino.AddPeers([]string{ h3.P2PAddress(), h2.P2PAddress(),h1.P2PAddress()}))
```
Comment (Roasbeef) [inline sync_test.go]: Style nit: we typically attempt to wrap all lines to 80 columns on a best effort basis.

**Comment by parakeety** [comment]
I fixed the styling. I'll send a PR to lnd 👍

File: neutrino.go [hunk 2]
```diff
@@ -477,6 +476,62 @@ type Config struct {
 	NameResolver func(host string) ([]net.IP, error)
 }
 
+// DataDir is a functional option that allows callers of NewChainService to set the
+// data directory for storing header information.
+func DataDir(dataDir string) func(*Config) {
+	return func(c *Config) {
+		c.DataDir = dataDir
```
Comment (halseth) [inline neutrino.go]: When adding functional options, I think we should unexport the set fields?

Comment (parakeety) [reply neutrino.go]: Good call. It's better to keep scope as small as possible. I'll change the fields to private

**Pull request closed without merging**